app/controllers/application_controller.rb

```diff
@@ -24,7 +24,7 @@ def confirm_change(object, message = nil)
 
   def report_errors(object)
     flash[:errors] = object.errors.full_messages
-    redirect_to :back
+    redirect_back fallback_location: 'public/404.html'
```

Comment:
Why a 404 page here?

Difficult to come up with a truly generic path here, so I might recommend another method argument.